### PR TITLE
Go Switch Statement

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

Added a switch statement for the somewhat empty Go section.  Lifted the entry from the Codecademy Docs Entry Assignment Sheet.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/go/concepts/switch/switch.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    <statements>
+    .
+    .
+    .
+default:
+  <statements>
+}
+```
+
+Values are compared left-to-right and top-to-bottom. The first one to match the `switch` expression determines the `case` `<statements>` block to be executed.
```
Comment (KTom101) [inline content/go/concepts/switch/switch.md:86]:
```suggestion
Values are compared left-to-right and top-to-bottom. The first `case` block to match the `switch` expression will be executed.
```

File: content/go/concepts/switch/switch.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+## Examples
+
+```go
+i := 7
+switch i % 2 {
+  case 0:
+     fmt.Printf("%d is even.\n",i)
+  case 1:
+     fmt.Printf("%d is odd.\n",i)
+}
```
Comment (KTom101) [inline content/go/concepts/switch/switch.md:97]:
```suggestion
}

// Output: 7 is odd.
```

File: content/go/concepts/switch/switch.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+```go
```
Comment (KTom101) [inline content/go/concepts/switch/switch.md:100]: This example might be useful as a codebyte, so the reader can see the output.
```suggestion
```codebyte/go
package main

import (
  "fmt"
  "time"
)
```

File: content/go/concepts/switch/switch.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+switch time.Now().Weekday() {
```
Comment (KTom101) [inline content/go/concepts/switch/switch.md:101]: adding main for codebyte
```suggestion
func main() {
  switch time.Now().Weekday() {
```

File: content/go/concepts/switch/switch.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  case time.Saturday, time.Sunday:
+    fmt.Println("It's the weekend")
+  case time.Friday:
+    fmt.Println("T.G.I.F.")
+  default:
+    fmt.Println("It's a weekday")
+}
```
Comment (KTom101) [inline content/go/concepts/switch/switch.md:108]: function ending bracket for main 
```suggestion
  }
}
```

**Review by KTom101 — COMMENTED** [review]
Hi @SSwiniarski, I've added my comments for this. Let me know what you think.

**Comment by SSwiniarski** [comment]
@KTom101, I applied all your changes.

**Comment by KTom101** [comment]
@SSwiniarski Great! Handing this over to @Dusch4593 or @yangc95!

**Review by KTom101 — APPROVED** [review]

File: content/go/concepts/switch/switch.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+case <value N>:
+  <statements>
+default:
+  <statements>
+}
+```
+
+When `<expression>` is equal to the value of one of the `case` statements, the corresponding `<statements>` block is executed. If the `<expression>` matches none of the `case` values, then the optional `default` `<statements>` block is executed.
+
+The `switch` statement in Go differs from other languages like C++ and Java, in that the values used by the `case` statements don't need to be constants, they can be variables or full expressions. Also, unlike in C ++ and Java, Go only executes the selected "case" statement block, so no "break" statement is needed.
```
Comment (yangc95) [inline content/go/concepts/switch/switch.md:40]: Can we link to some of these: https://www.google.com/search?q=codecademy+docs+switch&oq=codecademy+docs+switch&aqs=chrome..69i57j69i60l2.9087j0j1&sourceid=chrome&ie=UTF-8

File: content/go/concepts/switch/switch.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+## Syntax 2
+
+The expression can be omitted after the `switch` and the `switch` behaves more like an `if ... then ... else` statement:
```
Comment (yangc95) [inline content/go/concepts/switch/switch.md:44]:
```suggestion
The expression can be omitted after the `switch` making it behave more like an `if ... then ... else` statement:
```

**Review by yangc95 — COMMENTED** [review]
@SSwiniarski Review 2️⃣ fin. 👋

File: content/go/concepts/switch/switch.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+Tags:
+  - 'Comparison'
+  - 'Control Flow'
+  - 'Switch'
+CatalogContent:
+  - 'learn-go'
+  - 'paths/back-end-engineer-career-path'
+---
+
+A `switch` statement allows the selective execution of one of several blocks of code based on the value of an expression. It can be thought of as an expanded `if ... then ... else` statement, with many potential `then` clauses.
```
Comment (Dusch4593) [inline content/go/concepts/switch/switch.md:16]: Are we sure it's an `if...then...else` statement and not just `if..else`? I am searching the official Go docs and can't find that specific reference: https://go.dev/ref/spec#If_statements

Comment (KTom101) [reply content/go/concepts/switch/switch.md:16]: I think you're right @Dusch4593 . They compare it to an `if...else` statement on their "A Tour of Go" webpage: https://go.dev/tour/flowcontrol/9#:~:text=A%20switch%20statement%20is%20a,all%20the%20cases%20that%20follow.

File: content/go/concepts/switch/switch.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+  - 'paths/back-end-engineer-career-path'
+---
+
+A `switch` statement allows the selective execution of one out of several blocks of code based on the value of an expression. It can be thought of as a expanded `if ... then ... else` statement, with many potential `then` clauses.
+
+## Syntax 1
+
+This is the typical `switch` statement syntax:
+
+```pseudo
```
[the remark this answers is not in the document]
Comment (Dusch4593) [reply content/go/concepts/switch/switch.md:22]: ^^^ I agree here, it seems like this is also the case on in Go's ref spec for switch expressions ([under "Expression switches"](https://go.dev/ref/spec#Switch_statements#Expression_switches)

File: content/go/concepts/switch/switch.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  .
+  .
+  case <value N>:
+    <statements>
+  default:
+    <statements>
+}
+```
+
+When `<expression>` is equal to the value of one of the `case` statements, the corresponding `<statements>` block is executed. If the `<expression>` matches none of the `case` values, then the optional `default` `<statements>` block is executed.
```
Comment (Dusch4593) [inline content/go/concepts/switch/switch.md:38]: What happens if there is no `default` block? Will it cause an error or something?

Comment (Dusch4593) [inline content/go/concepts/switch/switch.md:38]: ~`default` `statements`~ -> `default statements`~

~`case` `statements`~ -> `case statements`~

File: content/go/concepts/switch/switch.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+The `switch` statement in Go differs from other languages like [C++](https://www.codecademy.com/resources/docs/cpp/switch) and [Java](https://www.codecademy.com/resources/docs/java/switch), in that the values used by the `case` statements don't need to be constants, they can be variables or full expressions. Also, unlike in C ++ and Java, Go only executes the selected "case" statement block, so no "break" statement is needed.
```
Comment (Dusch4593) [inline content/go/concepts/switch/switch.md:40]:
```suggestion
In Go, the `switch` statement differs from other languages like [C++](https://www.codecademy.com/resources/docs/cpp/switch) and [Java](https://www.codecademy.com/resources/docs/java/switch) in that the values used by the `case` statements are not limited to constants. They can also be variables or full expressions. Additionally, Go only executes the selected `case <statements> block, so no "break" statement is needed.
```

> Additionally, Go only executes the selected `case <statements>` block, so no "break" statement is needed. 

Does this mean that Go will keep moving down the program and exit the switch statement after a case is selected?

File: content/go/concepts/switch/switch.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+    .
+    .
+case <expression N>
+  <statements>
+default:
+  <statements>
+}
+```
+
+In this example, the first expression that evaluates true will execute the corresponding `case` `<statements>` block. Again, `default` is optional, and will execute its `<statements>` block if none of the `case` statements are executed.
```
Comment (Dusch4593) [inline content/go/concepts/switch/switch.md:62]:
```suggestion
In this example, the first `case <expression>` that evaluates to true will execute its corresponding `<statements>`. Again, the `default` block is optional, and will execute its `<statements>` if none of the `case` statements are executed.
```

File: content/go/concepts/switch/switch.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+func main() {
+  switch time.Now().Weekday() {
+  case time.Saturday, time.Sunday:
+    fmt.Println("It's the weekend")
+  case time.Friday:
+    fmt.Println("T.G.I.F.")
+  default:
+    fmt.Println("It's a weekday")
+  }
+}
```
Comment (Dusch4593) [inline content/go/concepts/switch/switch.md:119]: Could we separate the `switch` expression above it and into a variable?

```
func main() {
  today = time.Now().Weekday()
  
  switch today {
  case time.Saturday, time.Sunday:
    fmt.Println("It's the weekend")
  case time.Friday:
    fmt.Println("T.G.I.F.")
  default:
    fmt.Println("It's a weekday")
  }
}
```

Also, I noticed something weird when I add the line `fmt.Println(time.Now())`. This is what I get:

```
It's a weekday
2009-11-10 23:00:00 +0000 UTC m=+0.000000001
```

Weird, eh?

File: content/go/concepts/switch/switch.md (hunk 11 above)
Comment (Dusch4593) [reply content/go/concepts/switch/switch.md:40]: Never mind just confirmed that's the case (yay Go Playground!)

File: content/go/concepts/switch/switch.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+When `<expression>` is equal to the value of one of the `case` statements, the corresponding `<statements>` block is executed. If the `<expression>` matches none of the `case` values, then the optional `default` `<statements>` block is executed.
+
+The `switch` statement in Go differs from other languages like [C++](https://www.codecademy.com/resources/docs/cpp/switch) and [Java](https://www.codecademy.com/resources/docs/java/switch), in that the values used by the `case` statements don't need to be constants, they can be variables or full expressions. Also, unlike in C ++ and Java, Go only executes the selected "case" statement block, so no "break" statement is needed.
+
+## Syntax 2
+
+The expression can be omitted after the `switch` making it behave more like an `if ... then ... else` statement:
```
Comment (Dusch4593) [inline content/go/concepts/switch/switch.md:44]:
```suggestion
The expression after the `switch` can be omitted, making the `case`s behave more like `<expressions>` in `if... else` statements:
```

File: content/go/concepts/switch/switch.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+switch i % 2 {
+case 0:
+   fmt.Printf("%d is even.\n",i)
+case 1:
+   fmt.Printf("%d is odd.\n",i)
+}
+
+// Output: 7 is odd.
+```
+
```
Comment (Dusch4593) [inline content/go/concepts/switch/switch.md:101]:
```suggestion

## Codebyte Example

```

File: content/go/concepts/switch/switch.md [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+This is equivalent to:
+
+```pseudo
+switch true { ... }
+```
+
+## Syntax 3
+
+A given `case` statement can have more than one matching value, separated by commas:
```
Comment (Dusch4593) [inline content/go/concepts/switch/switch.md:72]: Is this only the case if we're working with a `switch <expression> {` as opposed to a regular `switch {`?

File: content/go/concepts/switch/switch.md [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+CatalogContent:
+  - 'learn-go'
+  - 'paths/back-end-engineer-career-path'
+---
+
+A `switch` statement allows the selective execution of one of several blocks of code based on the value of an expression. It can be thought of as an expanded `if ... then ... else` statement, with many potential `then` clauses.
+
+## Syntax 1
+
+This is the typical `switch` statement syntax:
```
Comment (Dusch4593) [inline content/go/concepts/switch/switch.md:20]:
```suggestion
There are many syntaxes that be applied in the `switch` statement. Probably the more common one looks like this:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
Hey @SSwiniarski! I took a look and left a few comments/suggestions for you to look at. After the next batch of changes, this entry should be clear to merge.

**Comment by SSwiniarski** [comment]
@Dusch4593, I applied all your suggestions with some modifications.

**Review by Dusch4593 — APPROVED** [review]
LGTM!

File: content/go/concepts/switch/switch.md [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+CatalogContent:
+  - 'learn-go'
+  - 'paths/back-end-engineer-career-path'
+---
+
+A `switch` statement allows the selective execution of one of several blocks of code based on the value of an expression. It can be thought of as an expanded `if ... else` statement, with many potential `if` clauses.
+
+## Syntax 1
+
+There are many syntaxes that can be used in the `switch` statement. The following is probably the most common syntax:
```
Comment (yangc95) [inline content/go/concepts/switch/switch.md:20]:
```suggestion
There are many syntaxes that can be used in the `switch` statement. The following is probably the most common:
```

File: content/go/concepts/switch/switch.md [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+  <statements>
+case <value 2>:
+  <statements>
+  .
+  .
+  .
+case <value N>:
+  <statements>
+default:
+  <statements>
```
Comment (yangc95) [inline content/go/concepts/switch/switch.md:34]:
```suggestion
  case <value 1>:
    <statements>
  case <value 2>:
    <statements>
  .
  .
  .
  case <value N>:
    <statements>
  default:
    <statements>
```

File: content/go/concepts/switch/switch.md (hunk 18 above)
Comment (Dusch4593) [reply content/go/concepts/switch/switch.md:20]: `switch` the last sentence for "The following are probably" and I think we got a deal! 😄

**Comment by SSwiniarski** [comment]
@yangc95, are we good to merge this?

File: content/go/concepts/switch/switch.md (hunk 19 above)
Comment (yangc95) [reply content/go/concepts/switch/switch.md]: whoops sorry about this, saw the proper reference

File: content/go/concepts/switch/switch.md [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+}
+```
+
+When `<expression>` is equal to the value of one of the case statements, the corresponding `<statements>` block is executed. If the `<expression>` matches none of the `case` values, then the optional `default` statements block is executed. If the `default` block is omitted, and none of the values are matched, none of the statements in the switch block are executed.
+
+In Go, the `switch` statement differs from other languages like [C++](https://www.codecademy.com/resources/docs/cpp/switch) and [Java](https://www.codecademy.com/resources/docs/java/switch) in that the values used by the `case` statements are not limited to constants. They can also be variables or full expressions. Additionally, Go only executes the selected `case <expressions>` block, or the `default` block, so no "break" statement is needed. Code execution automatically resumes after the `switch` statement.
```
Comment (yangc95) [inline content/go/concepts/switch/switch.md:40]:
```suggestion
In Go, the `switch` statement differs from other languages like [C++](https://www.codecademy.com/resources/docs/cpp/switch) and [Java](https://www.codecademy.com/resources/docs/java/switch) in that the values used by the `case` statements are not limited to constants. They can also be variables or full expressions. Additionally, Go only executes the selected `case <expressions>` block, or the `default` block, so no `break` statement is needed. Code execution automatically resumes after the `switch` statement.
```

**Review by yangc95 — APPROVED** [review]

**Comment by Dusch4593** [comment]
Nice, the entry is now live! https://www.codecademy.com/resources/docs/go/switch

I wonder why the snippet in the "Codebyte Example" section opens with this screen?

Screen Shot 2022-02-19 at 12 48 48 PM

**Comment by yangc95** [comment]
> I wonder why the snippet in the "Codebyte Example" section opens with this screen?
> 

On a side note, I really like this example. After getting through the drop-down and picking a language for the code byte, I got a nice little message.

Not sure why this is happening ```codebyte/go

@christian-dinh

**Pull request merged**